core/src/org/pentaho/di/core/gui/BaseFileInputFieldInterface.java

```diff
@@ -22,7 +22,7 @@
 
 package org.pentaho.di.core.gui;
 
-public interface TextFileInputFieldInterface extends Comparable<TextFileInputFieldInterface> {
```

Comment:
I'm very sorry, I didn't realize that part of this chain was a core class that has existed for a while.  For org.pentaho.di.core.gui.TextFileInputFieldInterface, please make a new one called BaseFileInputFieldInterface with the same signature, then change the existing TextFileInputFieldInterface to extend that one and deprecate it.  We can refactor any usage we find in pentaho-kettle, but that way we won't break any external usages of the interface.